### PR TITLE
Add functions to load selectors and highlight active page

### PR DESCRIPTION
**Pull request by BryanTriana**
The additions of these functions enable loading an HTML element from an external document as well as a feature to highlight the current active page when navigating the website. There is also minor stylistic changes to the CSS files and the addition of the required attribute for input tags.

### CODE REVIEW

File: portfolio/src/main/webapp/navbar.html [hunk 1]
```diff
@@ -1,5 +1,8 @@
 <!DOCTYPE html>
 <html>
+  <head>
+    <meta charset="UTF-8">
+  </head>
```
Comment (rharagutchi) [inline portfolio/src/main/webapp/navbar.html:5]: Out of curiosity, why is this needed?

Comment (BryanTriana) [reply portfolio/src/main/webapp/navbar.html:5]: It is normally good practice to put the encoding in the head tag of an HTML document. In the case of navbar.html I see that this is not required since it is just an html file that will get embedded in another one. In some of the changes I made today I actually removed the head tags for navbar.html and post-comment.html since they are not needed.

File: portfolio/src/main/webapp/script.js [hunk 2]
```diff
@@ -1,28 +1,30 @@
...
+ * @param { string } selectorID - ID that will be used across the document to refer to the HTML element
+ * @param { string } filename - Filename of HTML file that is used to load the element
+ *
+ * @example loadSelector("navbar", "navbar.html")
  */
-function addRandomGreeting() {
-  const greetings =
-      ['Hello world!', '¡Hola Mundo!', '你好，世界！', 'Bonjour le monde!'];
+function loadSelector(selectorID, filename) {
+    $(document).ready(function() {
```
Comment (codewithjess) [inline portfolio/src/main/webapp/script.js:11]: I believe you can still use jquery in the js file.

File: portfolio/src/main/webapp/script.js [hunk 3]
```diff
@@ -1,28 +1,30 @@
...
+function highlightActivePage() {
+    const linkContainer = document.getElementById('link-container');
+    const links = linkContainer.getElementsByTagName('a');
 
-  // Add it to the page.
-  const greetingContainer = document.getElementById('greeting-container');
-  greetingContainer.innerText = greeting;
+    for(let i = 0; i < links.length; ++i) {
+        if(links[i].href === window.location.href) {
+            links[i].classList.add('active');
```
Comment (codewithjess) [inline portfolio/src/main/webapp/script.js:27]: will there only be one active link at a given point - can you return early here?

Comment (BryanTriana) [reply portfolio/src/main/webapp/script.js:25]: Yes thanks for pointing that out, I made the change in the new commit

File: portfolio/src/main/webapp/navbar.html (hunk 1 above)
Comment (BryanTriana) [reply portfolio/src/main/webapp/navbar.html]: I removed the unnecessary tags on the latest commit.

File: portfolio/src/main/webapp/script.js (hunk 2 above)
Comment (BryanTriana) [reply portfolio/src/main/webapp/script.js:10]: Hi Jessica, I don't understand what you mean by that. Should I be using the jquery code in the HTML file instead of the JS file, or keep it like it currently is?

Comment (codewithjess) [reply portfolio/src/main/webapp/script.js:10]: Oops, I read this part incorrectly. This section is fine :)

Seems like `highlightActivePage()` can use jquery though, no?

`document.getElementsByTagName('nav');` => `$("nav");`
or you can combine the two getters: `const links = $("nav a");`

File: portfolio/src/main/webapp/script.js (hunk 3 above)
Comment (BryanTriana) [reply portfolio/src/main/webapp/script.js:24]: Oh wow I didn't know you could do that with jquery. I added this change in the new commit.

Comment (codewithjess) [reply portfolio/src/main/webapp/script.js:24]: nice! this code should also be wrapped in `$(document).ready(function(){ ... });` like above.

File: portfolio/src/main/webapp/script.js [hunk 4]
```diff
@@ -1,28 +1,28 @@
...
+ * @param { string } filename - Filename of HTML file that is used to load the element
+ *
+ * @example loadSelector("navbar", "navbar.html")
  */
-function addRandomGreeting() {
-  const greetings =
-      ['Hello world!', '¡Hola Mundo!', '你好，世界！', 'Bonjour le monde!'];
+function loadSelector(selectorID, filename) {
+    $(document).ready(function() {
+        $('#' + selectorID).load(filename);
```
Comment (codewithjess) [inline portfolio/src/main/webapp/script.js:11]: Instead of hardcoding "#", can callers pass in the full selector themselves?

This will make the function more flexible as it can take in any type of selector.

File: portfolio/src/main/webapp/script.js [hunk 5]
```diff
@@ -1,28 +1,28 @@
...
+ * if its children anchor links have the same 'href' attribute
+ */
+function highlightActivePage() {
+    const links = $('nav a');
 
-  // Add it to the page.
-  const greetingContainer = document.getElementById('greeting-container');
-  greetingContainer.innerText = greeting;
+    for(let i = 0; i < links.length; ++i) {
+        if(links[i].href === window.location.href) {
```
Comment (rharagutchi) [inline portfolio/src/main/webapp/script.js:23]: Usually we use spaces after things like "for" and "if" statements
```suggestion
    for (let i = 0; i < links.length; ++i) {
        if (links[i].href === window.location.href) {
```

File: portfolio/src/main/webapp/blog.html [hunk 6]
```diff
@@ -2,110 +2,114 @@
...
-        std::mt19937</a>:
-        random number engine based on the 
-        <a href="https://en.wikipedia.org/wiki/Mersenne_Twister">
-        Mersenne Twister
-        </a>
-        algorithm.
-      </li>
-      <li>
+        href="https://en.cppreference.com/w/cpp/numeric/random/rand">rand()
+        </a>;
```
Comment (codewithjess) [inline portfolio/src/main/webapp/blog.html:27]: When line-wrapping, each continuation line should be indented at least 4 additional spaces from the original line.

[go/htmlcssstyle#HTML_Line-Wrapping](go/htmlcssstyle#HTML_Line-Wrapping)

File: portfolio/src/main/webapp/blog.html [hunk 7]
```diff
@@ -2,114 +2,114 @@
...
   <head>
     <meta charset="UTF-8">
-    <title>Bryan Triana - Blog</title>
+    <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
+    <link rel="stylesheet" href="https://stackpath.bootstrapcdn.com/bootstrap/4.3.1/css/bootstrap.min.css">
     <link rel="stylesheet" href="style.css">
-    <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.5.1/jquery.min.js"><script>
+    <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.5.1/jquery.min.js"></script>
+    <script src="https://cdnjs.cloudflare.com/ajax/libs/popper.js/1.14.7/umd/popper.min.js"></script>
+    <script src="https://stackpath.bootstrapcdn.com/bootstrap/4.3.1/js/bootstrap.min.js"></script>
```
[the remark this answers is not in the document]
Comment (BryanTriana) [reply portfolio/src/main/webapp/blog.html:10]: Ok, So do you think adding a function like the following and then calling it at the top of script.js file is a good way of avoiding this kind of imports?
```
function loadScripts(scripts) {
    for (let i = 0; i < scripts.length; ++i) {
        $head.append("<script src=\"" + scripts[i] + "\" type=\"text/javascript\"></script>")
    }
}
```

File: portfolio/src/main/webapp/blog.html (hunk 6 above)
Comment (BryanTriana) [reply portfolio/src/main/webapp/blog.html:27]: OK I don't seem to have access to that go link, but i'll add spaces for better indentation.

Comment (codewithjess) [reply portfolio/src/main/webapp/blog.html:27]: Hmm thats too bad, here's another link that should be public. https://google.github.io/styleguide/htmlcssguide.html#HTML_Line-Wrapping
